googleanalytics/google_analytics_report_tool.py

```diff
@@ -0,0 +1,109 @@
+import os
+from superagi.tools.base_tool import BaseTool
+from pydantic import BaseModel, Field
+import json
+from typing import Type, Optional
+from superagi.resource_manager.file_manager import FileManager
+import yaml
+from google.analytics.data_v1beta import BetaAnalyticsDataClient
+from google.analytics.data_v1beta.types import (
+    DateRange,
+    Dimension,
+    Metric,
+    RunReportRequest,
+)
+
+class GoogleAnalyticsReportToolInput(BaseModel):
+    start_date_of_query: str = Field(..., description="The starting date of the query, in YYYY-MM-DD format")
+    end_date_of_query: str = Field(..., description="The last date of the query, in YYYY-MM-DD format, if today, return today's date")
+    store_report_in_single_file: bool= Field(..., description="True if it is specified that the report should be stored in a single file. False otherwise.")
+
+
+class GoogleAnalyticsReportTool(BaseTool):
+    """
+    Analytics Report Tool
+    """
+    name: str = "Google Analytics Report Tool"
+    args_schema: Type[BaseModel] = GoogleAnalyticsReportToolInput
+    description: str = "Return a google analytics report for the information the user requires"
+    resource_manager: Optional[FileManager] = None
+
+    def _execute(self, start_date_of_query: str, end_date_of_query: str, store_report_in_single_file: bool):
+        property_id = int(self.get_tool_config("PROPERTY_ID"))
+        google_credentials = self.get_tool_config("GOOGLE_CREDENTIALS_FILE")
+
+        self._set_google_credentials(google_credentials)
+
+        client = BetaAnalyticsDataClient()
+        dimensions_and_metrics = self.get_dimensions_and_metrics()
+
+        filenames = ["report.txt"] if store_report_in_single_file else []
+        report=""
+
+        for dimensions, metrics in dimensions_and_metrics:
+            request = self._create_run_report_request(property_id, dimensions, metrics, start_date_of_query, end_date_of_query)
+            google_response = client.run_report(request)
```

Comment:
Remove   client = BetaAnalyticsDataClient() and directly do BetaAnalyticsDataClient().run_report(request)

Reply:
Temporary variables should be avoided it makes functions long and complex